File: src/cpp/mayer/lennard_jones.cpp
#include "lennard_jones.h"

#include <cmath>

LennardJones::LennardJones(double sigma, double epsilon, double k, double T)
    : sigma_(sigma), epsilon_(epsilon), k_(k), T_(T) {}

double LennardJones::uij(double rij) const {
    double ratio = sigma_ / rij;
    double ratio6 = std::pow(ratio,6);
    return 4.0 * epsilon_ * (ratio6 * ratio6 - ratio6);
}

double LennardJones::fij(double rij) const {
    return std::exp(-uij(rij) / (k_ * T_)) - 1.0;
}

File: src/msmc/mayer/lennard_jones.py
import msmc._core as _core


class LennardJones(_core.LennardJones):
    """Lennard-Jones pair potential.

    Parameters
    ----------
    sigma : float
        Length scale (distance at which potential is zero).
    epsilon : float
        Energy scale (depth of potential well).
    k : float
        Boltzmann constant.
    T : float
        Temperature.
    """

    def __init__(self, sigma, epsilon, k, T):
        super().__init__(sigma, epsilon, k, T)

    def uij(self, rij):
        """
        Lennard-Jones pair energy.

        Parameters
        ----------
        rij : float
            Separation distance between molecule i and j.

        Returns
        -------
        float
            Pair potential energy.
        """
        return super().uij(rij)
